accounts/models.py

```diff
@@ -350,55 +381,123 @@ def num_sounds_pending_moderation(self):
+                    user=self.user,
+                    username=self.user.username,
+                    email=self.user.email,
+                    date_joined=self.user.date_joined,
+                    last_login=self.user.last_login,
+                    reason=deletion_reason)
+
+            # If UserGDPRDeletionRequest object(s) exist for that user, update their status and deleted_user property
+            UserGDPRDeletionRequest.objects.filter(user_id=self.user.id)\
+                .update(status="de", deleted_user=deleted_user_object)
+
+            # If Sound/Pack objects will be deleted, we need to make sure that num_sound_downloads and
+            # num_pack_downloads fileds from users that downloaded packs or sounds from the user being delted are
+            # correct. Otherwise this will make the deletion process to fail (violating DB constraints)
+            if delete_user_object_from_db or remove_sounds:
+                sound_downloaders = Download.objects.filter(sound__user=self.user).only('user')
+                pack_downloaders = PackDownload.objects.filter(pack__user=self.user).only('user')
+                for download in sound_downloaders:
+                    download.user.profile.num_sound_downloads = Download.objects.filter(user=download.user).count()
+                    download.user.profile.save()
+                for download in pack_downloaders:
+                    download.user.profile.num_pack_downloads = PackDownload.objects.filter(user=download.user).count()
+                    download.user.profile.save()
+
+            if delete_user_object_from_db:
+                # If user is to be completely deleted from the DB, use delete() method. This will remove all
+                # related objects like sounds, packs, comments, etc...
+                self.user.delete()
+
+            else:
+                # If user object is not to be deleted from DB we need to anonymize it and remove sounds if requested
+
+                # Remove personal data from the user
+                self.user.username = 'deleted_user_%s' % self.user.id
+                self.user.email = 'deleted_user_%s@freesound.org' % self.user.id
+                self.has_avatar = False
+                self.is_anonymized_user = True
+                self.user.set_unusable_password()
+
+                self.about = ''
+                self.home_page = ''
+                self.signature = ''
+                self.geotag = None
+
+                self.save()
+                self.user.save()
+
+                # Remove existing OldUsername objects so there are no redirects to the anonymized/deleted user page
+                OldUsername.objects.filter(user=self.user).delete()
+
+                # Remove sounds and packs if requested
+                if remove_sounds:
+                    Sound.objects.filter(user=self.user).delete()
+                    Pack.objects.filter(user=self.user).update(is_deleted=True)
```

Comment:
why do we actually delete the sounds, but only mark the packs as deleted with a flag?

Reply:
didn't see a reply from you about this

Reply:
this discussion is happening outside this PR. could be something to consider for the delete sounds refactor that should come after this PR